src/go/configinfo/service_info.go

```diff
@@ -891,3 +922,16 @@ func calculateStreamIdleTimeout(operationDeadline time.Duration, opts options.Co
 	operationIdleTimeout := operationDeadline + time.Second
 	return util.MaxDuration(operationIdleTimeout, opts.StreamIdleTimeout)
 }
+
+func parseRetriableStatusCodes(statusCodes string) ([]uint32, error) {
+	codeList := strings.Split(statusCodes, ",")
+	var codes []uint32
+	for _, codeStr := range codeList {
+		if code, err := strconv.Atoi(codeStr); err != nil || code < 100 || code >= 600 {
+			return nil, fmt.Errorf("invalid http status codes: %v, the valid one should be a number in [100, 600)", code)
```

Comment:
nit: put `err` in the message

Reply:
This error `strconv.Atoi: parsing "invalid-status-code": invalid syntax` is not that informative   and is covered in `the valid one should be a number in [100, 600)`. So prefer not.